docs/advanced/UsageWithReactRouter.md

```diff
@@ -1,4 +1,162 @@
 # Usage with React Router
 
-Sorry, but we're still writing this doc.  
-Stay tuned, it will appear in a day or two.
+So you want to do routing with your Redux app. You can use it with [React Router](https://github.com/reactjs/react-router). Redux will be the source of truth for your data and React Router will be the source of truth for your URL. In most of the cases, **it is fine** to have them separate unless if you need to time travel and rewind actions that triggers the change URL.
+
+## Installing React Router
+`react-router` is available on npm :
+
+`npm install --save react-router`
+
+## Configuring the Fallback URL
+
+Before implementing React Router, we need to configure our development server. Indeed, our development server is not currently aware of the declared routes in React Router. If you refresh your page or try to access directly an URL declared in React Router without having configured your development server, you will get a 404 HTTP error. You will be first requesting a url that the development server is not aware of, instead of asking to React Router. You need to configure a fallback URL, to serve index.html on an unknown URL so that in the front-end, React Router can handle the request.
+
+>##### Note on Create React App
+
+> If you are using Create React App, you won't need to configure a fallback URL, it is automatically done.
+
+### Configuring Express.js
+If you are serving your index.html from Express.js :
+``` js
+  app.get('/*', (req,res) => {
+    res.sendfile(path.join(__dirname, 'index.html'))
+  })
+```
+
+### Configuring Webpack Dev Server
+If you are serving your index.html from Webpack Dev Server:
+You can add to your webpack.config.dev.js :
+```js
+  devServer: {
+    historyApiFallback: true,
+  }
+```
+
+## Connecting React Router with Redux App
+
+Along this chapter, we will be using the [Todos](https://github.com/reactjs/redux/tree/master/examples/todos) example. We recommend you to clone it while reading this chapter.
+
+The `<Router />` component has to be a children of `<Provider />` so that route handlers can get access to the `store`. `<Provider />` is the higher-order component provided by react-redux that lets you bind Redux to React (see [Usage with React](../basics/UsageWithReact.md)).
```

Comment:
Instead of "has to be children" let's be more conversational. Like "We will wrap ... in ... because ..."